.github/workflows/run-tests.yml

```diff
@@ -487,12 +487,24 @@ jobs:
           annotate_only: true
           skip_annotations: true
 
+      - name: Get job ID from workflow run
+        id: get_job_id
+        run: |
+          set -e
+          job_id=$(curl -L \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{secrets.GITHUB_TOKEN}}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            "https://api.github.com/repos/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}/jobs" \
+            | jq "first(.jobs[] | select(.name | endswith(\"$job_name\")) | .id)")
+          echo "JOB_ID=${job_id}" >> "$GITHUB_OUTPUT"
```

Comment:
Personally I'd probably store this as a shell script in the repo to avoid repeating it 3 times (and for potential re-use since this has now come up in `temporal` and `saas-cicd`). (But not blocking.)

Reply:
you right, it is the same...sigh